### PR TITLE
add main page

### PR DESCRIPTION
**Pull request by dolphin323**

### CODE REVIEW

File: frontend/src/reducers/projects.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import { Project } from '../common/types';
```
Comment (yevhenii-moroziuk) [inline frontend/src/reducers/projects.ts]: Let's move it to mainPage scene

File: frontend/src/scenes/MainPage/index.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        <div className={classes['main-text']}>
+          <div className={classes['main-logo-text']}>
+            <div>
+              <span className={classes.logo}>
+                HypeCrafter
+              </span> faithful assistant
+            </div>
+          </div>
+          <div className={classes['logo-text']}>
+            Do you have a good and well thought-out idea and are you ready
```
Comment (yevhenii-moroziuk) [inline frontend/src/scenes/MainPage/index.tsx]: Use TFunction please. The key can be: "Do you have a good and well thought-out idea and are you ready_"

File: frontend/src/scenes/MainPage/index.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            to conquer the peaks on your own to make your dream come true?
+            But sometimes, alone, the difficult path from concept to finished
+            product is not easy at all. HypeCrafter will lend a helping hand,
+            and more than one!
+          </div>
+          <div className={classes.buttons}>
+            <Button type="button">Create Project</Button>
+            <Button type="button" isOutline>Help Project</Button>
```
Comment (yevhenii-moroziuk) [inline frontend/src/scenes/MainPage/index.tsx]: just outline

File: frontend/src/components/Button/styles.module.scss [hunk 4]
```diff
@@ -23,6 +27,9 @@
 }
 
 .outline {
+  border: solid 2px transparent;
```
Comment (yevhenii-moroziuk) [inline frontend/src/components/Button/styles.module.scss:30]: What if I select 

`<Button type="secondary" outline />`

?

**Review by yevhenii-moroziuk — CHANGES_REQUESTED** [review]

File: frontend/src/scenes/MainPage/index.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    popular, recommended
+  }));
+
+  useEffect(() => {
+    dispatch(fetchRecommendedProjectsAction());
+    dispatch(fetchPopularProjectsAction());
+  }, [dispatch]);
+
+  return (
+    <div className={classes.root}>
```
Comment (yevhenii-moroziuk) [inline frontend/src/scenes/MainPage/index.tsx:26]: It would be nice to use not only divs

**Review by yevhenii-moroziuk — CHANGES_REQUESTED** [review]

**Review by mtseluiko — CHANGES_REQUESTED** [review]
resolve conflicts

File: frontend/src/scenes/MainPage/index.tsx [hunk 6]
```diff
@@ -80,37 +79,41 @@ const MainPage: FC = () => {
...
-                goal={project.goal}
-                percent={calcDonationProgress(project.donated, project.goal)}
-                image="https://source.unsplash.com/random/800x600"
-              />
-            ))}
+            <LoaderWrapper isLoading={isStartupsLoading}>
+              {popularStartups.map((project: Project) => (
+                <ProjectCard
+                  key={project.id}
+                  to="/"
```
Comment (yevhenii-moroziuk) [inline frontend/src/scenes/MainPage/index.tsx]: to={`/projects/${project.id}`}
...
image={project.image}

File: backend/src/data/repositories/project.ts [hunk 7]
```diff
@@ -3,56 +3,59 @@ import { Project } from '../entities/project';
...
-    description, 
-    project.name, 
-    project."id", 
-    goal, 
-    category.name 
-  ORDER BY project."totalInteractionTime"/project."totalViews"/project."minutesToRead" DESC
-LIMIT 3
-    `);
+    return this.createQueryBuilder('project')
+      .select(`amount as donated,
```
Comment (yevhenii-moroziuk) [inline backend/src/data/repositories/project.ts]: Formatting

**Review by yevhenii-moroziuk — CHANGES_REQUESTED** [review]

**Review by yevhenii-moroziuk — APPROVED** [review]

File: frontend/src/scenes/MainPage/index.tsx [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+      { mainPage }
+    ) => ({
+      popular: mainPage.popular,
+      recommended: mainPage.recommended,
+      isLoading: mainPage.isLoading,
+      topics: mainPage.topics
+    })
+  );
+  useEffect(() => {
+    dispatch(actions.fetchPopularAndRecommendedProjectsAction());
```
Comment (yevhenii-moroziuk) [inline frontend/src/scenes/MainPage/index.tsx]: redundant dispatch

**Review by yevhenii-moroziuk — CHANGES_REQUESTED** [review]

**Review by yevhenii-moroziuk — APPROVED** [review]

**Pull request merged**